lib/DDG/Spice/Symbolab.pm

```diff
@@ -114,25 +117,25 @@ sub isLegal{
 			next;
 		}
 		
-		# i.e x^2, x^{2}
-		if ($word =~ /[xtu]\^(\{)?[xtu\d+](\})?/){
+		# i.e x^2, x^{2}, 2^{x}
+		if ($word =~ /([xtu]\d+)\^\{?([xtu]|\d+)\}?/){
 			$minWeight -= 3;
 			next;
 		}
 		
 		# i.e 5i
-		if ($word =~ /[\d]i/){
+		if ($word =~ /\di/){
 			$minWeight -= 3;
 			next;
 		}
 		
 		# i.e 'i'
-		if ($word eq 'i'){
+		else if ($word eq 'i'){
 			$minWeight -= 1;
 			next;
 		}
 		
-		if ($word !~ /\w\w\w\w/ and ($word =~ /[\^\*\=\+\/\\\|]/ or $word =~ /^\d+x$/)){
+		if ($word !~ /\w{4}/ and ($word =~ /[^*=+/\|]/ or $word =~ /^\d+x$/)){
```

Comment:
sorry in this case, the `^` **does** need to be escaped if it's _first_ in the character class, because `[^ ... ]` is actually a negated character class:

- `[^*=+/\|]` = accept any character that ISN'T `*,=,+,/,\, or |`
- `[*^=+/\|]` = accept any character that IS `*,^, =,+,/,\, or |`